### ISSUE TITLE
useCombobox with menu-trigger=focus and aria-labelledby shouldn't apply aria-hidden to the label element

### ISSUE THREAD

**Issue by ashleyryan** [adobe/react-spectrum#8768]
### Provide a general summary of the issue here

When the combobox popover opens, everything else on the page has aria-hidden applied to it to, to make sure it's ignored by screenreaders. However, when using aria-labelledby on the combobox input element, combined with menuTrigger=focus, this means that the name of the input isn't announced to screenreaders because the popover opens instantly and the label is aria-hidden.

### 🤔 Expected Behavior?

If there is an aria-labelledby, elements with those ids do not have aria-hidden applied

### 😯 Current Behavior

aria-hidden is applied to all elements, including the label element, resulting in the input element no longer being labelled.

### 💁 Possible Solution

We want to be able to use this component alongside our FormField component which has a separate label element instead of requiring `aria-label`. 

Without digging much into the code, the utility that applies aria-hidden to everything under the overlay could potentially also accept a list of ids to skip. perhaps `useComboboxState` could return that in the `state` it returns , which is used as the overlayState

### 🔦 Context

This is an accessibility violation

### 🖥️ Steps to Reproduce

https://codesandbox.io/p/sandbox/hardcore-moon-xzc4r

Notice that when focus is applied to the input the popover opens and applies aria-hidden to the associated label

### Version

3.21.0

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

OSX

**Comment by snowystinger** [comment]
I don't think that aria-hidden will stop the aria-labelledby. I can see the computed property for the label is still "Favorite Animal" even though the label is aria-hidden.

<img width="1509" height="946" alt="Image" src="https://github.com/user-attachments/assets/8283ce48-b632-4a57-92d4-2ed0be491bac" />

Did you have a real life issue with it?

If you did something like
```
      <label id="combobox-label">
        Favorite <div aria-hidden="true">Animal</div>
      </label>
```

then that would hide "Animal", but if you directly reference a hidden element with an aria-labelledby, then it does not consider the aria-hidden on it

**Comment by ashleyryan** [comment]
False alarm, you're correct. MDN even says `aria-labelledby` can reference hidden elements:

> The aria-labelledby property value can include content from elements that aren't even visible. While you should provide assistive technology users with the same content as all other users, you can include content from elements with the HTML [hidden](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Global_attributes/hidden) attribute, CSS [display: none](https://developer.mozilla.org/en-US/docs/Web/CSS/display), and CSS [visibility: hidden](https://developer.mozilla.org/en-US/docs/Web/CSS/visibility) in the calculated name string.

I can hear the label being announced in VoiceOver, so this must have been our auditors sharing one of their "best practices" rather than strict rules. They originally flagged one of our consumer's implementation because it wasn't labelled at all, and then during the validation testing, sent it back again because of the aria-hidden.

Sorry about that.
